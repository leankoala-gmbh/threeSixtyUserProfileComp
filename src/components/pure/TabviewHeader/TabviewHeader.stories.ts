import type { Meta, StoryObj } from '@storybook/vue3'
import TabviewHeader from './TabviewHeader.vue'

const meta: Meta<typeof TabviewHeader> = {
  title: 'Components/Pure/TabviewHeader',
  component: TabviewHeader,
  tags: ['autodocs'],
  argTypes: {
    // Define your interactive Props here
    // example: {
    //  options: ['text', 'password', 'email', 'number'],
    //  control: {
    //    type: 'select'
    //  }
  },
  parameters: {
    docs: {
      description: {
        component: 'A simple header element for tabviews'
      }
    }
  }
}

export default meta
type Story = StoryObj<typeof TabviewHeader>

export const TabviewHeaderStory: Story = {
  // name: 'TabHeader with subline',
  render: (args) => ({
    components: { TabviewHeader },
    setup() {
      return { args }
    },
    template: '<TabviewHeader>{{ args.default }}<template #subline>{{args.subline}}</template></TabviewHeader>'
  }),
  args: {
    // Define the Data for your props
    default: 'Hello World',
    subline: 'This is a subline'
  }
}
