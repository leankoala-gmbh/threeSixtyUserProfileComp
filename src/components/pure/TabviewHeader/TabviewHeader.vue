<template>
  <header class="tabviewHeader mb-4">
    <h3 class="font-medium mb-1 text-lg">
      <slot />
    </h3>
    <p v-if="$slots.subline">
      <slot name="subline" />
    </p>
  </header>
</template>
