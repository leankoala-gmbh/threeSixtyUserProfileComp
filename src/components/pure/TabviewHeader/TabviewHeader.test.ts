import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import TabviewHeader from './TabviewHeader.vue'

describe('TabviewHeader.vue', () => {
  it('should render', () => {
    const wrapper = mount(TabviewHeader, {
      slots: {
        default: 'Hello World',
        subline: 'This is a subline'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
