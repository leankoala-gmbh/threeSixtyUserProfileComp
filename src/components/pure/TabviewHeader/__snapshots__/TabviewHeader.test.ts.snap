// Vitest Snapshot v1

exports[`TabviewHeader.vue > should render 1`] = `
"<header class=\\"tabviewHeader mb-4\\">
  <h3 class=\\"font-medium mb-1 text-lg\\">Hello World</h3>
  <p>This is a subline</p>
</header>"
`;
